lib/metrics/Histogram.js

```diff
@@ -61,6 +62,54 @@ Histogram.prototype.percentiles = function (percentiles) {
   return results;
 };
 
+Histogram.prototype.weightedPercentiles = function (percentiles) {
```

Comment:
I see that this is not being included in toJSON(), which as far as I know is the main mechanism of getting data from a metric in this library.

I also see that this method does some heavy lifting and would have performance impacts if included in the JSON output.

How are you thinking people use this functionality?

Reply:
- [ ] Update the README

Reply:
@felixge [suggested a non-breaking change gets merged easier](https://github.com/felixge/node-measured/pull/28#issuecomment-295609745) hence the new method. In my original PR I just replaced the percentiles method.

We were thinking of simply writing our own `toJSON` using this new `weightedPercentiles` method as we were not using everything from the original one anyway (e.g. mean/stddev that we didn't need for our non-normally distributed data e.g. latencies that we were using it for)

Reply:
Thinking out-loud 

What if the constructor properties object of a Histogram was updated to in addition to taking the sample to use also took a call back that performed the logic required for toJSON.

It could the default could be the current logic but you could choose to overwrite it.

```javascript
const histogramExample = new Histogram({
    percentileMethod: Histogram.wieghtedPercentiles
})
```


```javascript
function Histogram(properties) {
  properties = properties || {};

  this._sample    = properties.sample || new EDS();
  this._percentileMethod = properties.percentileMethod || this.precentiles
  this._min       = null;
  this._max       = null;
  this._count     = 0;
  this._sum       = 0;

  // These are for the Welford algorithm for calculating running variance
  // without floating-point doom.
  this._varianceM = 0;
  this._varianceS = 0;
}

...
...

Histogram.prototype.toJSON = function () {
  var percentiles = this._percentileMethod([0.5, 0.75, 0.95, 0.99, 0.999]);

  return {
    min      : this._min,
    max      : this._max,
    sum      : this._sum,
    variance : this._calculateVariance(),
    mean     : this._calculateMean(),
    stddev   : this._calculateStddev(),
    count    : this._count,
    median   : percentiles[0.5],
    p75      : percentiles[0.75],
    p95      : percentiles[0.95],
    p99      : percentiles[0.99],
    p999     : percentiles[0.999]
  };
};
```

Reply:
Actually that makes perfect sense 👍 (Probably I also didn't bother since we used a custom `toJSON` anyway but happy to make this change)

Reply:
@csabapalfi Can you update the following so we can get this merged and released.

- [ ] Update Histogram's constructor options to have a percentile function override, and make toJSON() use the override or default to the current logic.

- [ ] Update README to reflect the new option

- [ ] Bump minor version number in package.json

Thanks!

Reply:
That command sounds fine to me, on my other OSS projects with Travis we do releases via the Github UI https://github.com/yaorg/node-measured/releases/new and that automatically creates the tag, triggers Travis and via https://github.com/Nike-Inc/cerberus-java-client/blob/master/.travis.yml#L9 causes the release to happen.

I think running that command and having a line similar to that will cause travis to release to NPM. I will merge this now and work on wiring this to Travis again under the new org. So don't make the tag yet.